src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -325,8 +331,18 @@ def enable_local_git(resource_group_name, name, slot=None):
     return {'url' : _get_local_git_url(client, resource_group_name, name, slot)}
 
 def sync_site_repo(resource_group_name, name, slot=None):
-    return _generic_site_operation(resource_group_name, name, 'sync_repository',
-                                   slot)
+    try:
+        return _generic_site_operation(resource_group_name, name, 'sync_repository', slot)
+    except CloudError as ex:
+        raise _extract_real_error(ex)
+
+# webapp service's error payload doesn't follow ARM's error format, so we had to sniff out
+def _extract_real_error(ex):
+    try:
+        err = json.loads(ex.response.text)
+        return CLIError(err['Message'])
+    except:  # pylint: disable=bare-except
```

Comment:
Suggest `except Exception` instead of bare except.
http://stackoverflow.com/questions/18982610/difference-between-except-and-except-exception-as-e-in-python